crates/sui-rosetta/src/operations.rs

```diff
@@ -135,6 +139,38 @@ impl Operations {
         })
     }
 
+    fn pay_coin_ops_to_internal(self) -> Result<InternalOperation, Error> {
+        let mut recipients = vec![];
+        let mut amounts = vec![];
+        let mut sender = None;
+        let mut currency = None;
+        for op in self {
+            if let (Some(amount), Some(account)) = (op.amount.clone(), op.account.clone()) {
+                currency = currency.or(Some(amount.currency));
+                if amount.value.is_negative() {
+                    sender = Some(account.address)
+                } else {
+                    recipients.push(account.address);
+                    let amount = amount.value.abs();
+                    if amount > u64::MAX as i128 {
+                        return Err(Error::InvalidInput(
+                            "Input amount exceed u64::MAX".to_string(),
+                        ));
+                    }
+                    amounts.push(amount as u64)
+                }
+            }
+        }
+        let sender = sender.ok_or_else(|| Error::MissingInput("Sender address".to_string()))?;
```

Comment:
is that the only way we can find the sender?

Reply:
yep, remember how the incoming request payload looks like
```
[{
            "operation_identifier":{"index":0},
            "type":"PayCoin",
            "account": { "address" : "0x..." },
            "amount" : {
                "value": "30000000",
                "currency": {
                    "symbol": "TEST_COIN",
                    "decimals": 6,
                    "metadata": {
                        "coin_type": "0x...",
                    }
                }
            },
        },
        {
            "operation_identifier":{"index":1},
            "type":"PayCoin",
            "account": { "address" : "0x..." },
            "amount" : {
                "value": "-30000000",
                "currency": {
                    "symbol": "TEST_COIN",
                    "decimals": 6,
                    "metadata": {
                        "coin_type": "0x...",
                    }
                }
            },
        }]
```
which is in alignment to the existing PaySui operation. I wouldn't like to make this work in a different way at this point but something to be improved for both operations in a later point, if possible though, because it will most likely imply changes on the caller's side too.